tests/testthat/test-do-spark.R

```diff
@@ -11,6 +11,13 @@ register_test_spark_connection <- function() {
 
 register_test_spark_connection()
 
+y <- 3
+z <- 4
+fn <- function(x) { x * x - y }
+test_fn <- function(x) {
+  fn(fn(x * y) - z) - y * z
```

Comment:
but if `test_fn` were to contain an inner function, e.g.,
```
test_fn <- function(x) {
  inner_fn <- function(x) { ... }
   ...
}
```
, then this would still break and there would be an error saying 'function "function" not found' -- I believe that may be a parsing bug from `globals`. We may need to live with this limitation for now unfortunately.